File: WorldFeed/Server/src/Services/Upload/WorldFeed.Upload.API/Events/TweetEventArgs.cs
﻿//namespace WorldFeed.Upload.Events
//{
//    using System;

//    using WorldFeed.Common.Public.Models.Interfaces;
//    using WorldFeed.Upload.Domain;

//    public class TweetEventArgs : EventArgs
//    {
//        public TweetEventArgs(ITweet tweet, string json)
//        {
//            Tweet = tweet;
//            Json = json;
//        }

//        public ITweet Tweet { get; }
//        public string Json { get; }
//    }

//    /// <summary>
//    /// Event informing that a tweet was received by a stream
//    /// </summary>
//    public class TweetReceivedEventArgs : TweetEventArgs
//    {
//        public TweetReceivedEventArgs(ITweet tweet, string json) : base(tweet, json)
//        {
//        }
//    }

//    /// <summary>
//    /// Event informing that a tweet matching a stream criteria has been received
//    /// </summary>
//    public class MatchedTweetReceivedEventArgs : TweetEventArgs
//    {
//        public MatchedTweetReceivedEventArgs(ITweet tweet, string json) : base(tweet, json)
//        {
//        }

//        public string[] MatchingTracks { get; set; }
//        public ILocation[] MatchingLocations { get; set; }
//        public long[] MatchingFollowers { get; set; }
//        public MatchOn MatchOn { get; set; }

//        public string[] QuotedTweetMatchingTracks { get; set; }
//        public ILocation[] QuotedTweetMatchingLocations { get; set; }
//        public long[] QuotedTweetMatchingFollowers { get; set; }
//        public MatchOn QuotedTweetMatchOn { get; set; }
//    }

//    public class TweetLocationDeletedEventArgs : EventArgs
//    {
//        public TweetLocationDeletedEventArgs(ITweetLocationRemovedInfo tweetLocationRemovedInfo)
//        {
//            TweetLocationRemovedInfo = tweetLocationRemovedInfo;
//        }

//        public ITweetLocationRemovedInfo TweetLocationRemovedInfo { get; }
//    }

//    public class TweetWitheldEventArgs : EventArgs
//    {
//        public TweetWitheldEventArgs(ITweetWitheldInfo tweetWitheldInfo)
//        {
//            TweetWitheldInfo = tweetWitheldInfo;
//        }

//        public ITweetWitheldInfo TweetWitheldInfo { get; }
//    }
//}

File: Sharebook/Client/src/app/core/Public/Events/TweetEventArgs.ts
import {ITweet} from "../Models/Interfaces/ITweet";
import {ILocation} from "../Models/Interfaces/ILocation";
import {MatchOn} from '../Streaming/MatchOn';
import {ITweetLocationRemovedInfo} from "../Streaming/Events/ITweetLocationRemovedInfo";
import {ITweetWitheldInfo} from '../Streaming/Events/ITweetWitheldInfo';

export class TweetEventArgs /*extends EventArgs*/ {
  constructor(tweet: ITweet, json: string) {
    // super();

    this.tweet = tweet;
    this.json = json;
  }

  public tweet: ITweet;
  public json: string;
}

// Event informing that a tweet was received by a stream
export class TweetReceivedEventArgs extends TweetEventArgs {
  constructor(tweet: ITweet, json: string) {
    super(tweet, json);
  }
}

// Event informing that a tweet matching a stream criteria has been received
export class MatchedTweetReceivedEventArgs extends TweetEventArgs {
  constructor(tweet: ITweet, json: string) {
    super(tweet, json);
  }

  public matchingTracks: string[];
  public matchingLocations: ILocation[];
  public matchingFollowers: number[];
  public matchOn: MatchOn;

  public retweetMatchingTracks: string[];
  public retweetMatchingLocations: ILocation[];
  public retweetMatchingFollowers: number[];
  public retweetMatchOn: MatchOn;

  public quotedTweetMatchingTracks: string[];
  public quotedTweetMatchingLocations: ILocation[];
  public quotedTweetMatchingFollowers: number[];
  public quotedTweetMatchOn: MatchOn;
}

export class TweetLocationDeletedEventArgs /*extends EventArgs*/ {
  constructor(tweetLocationRemovedInfo: ITweetLocationRemovedInfo) {
   // super();

    this.tweetLocationRemovedInfo = tweetLocationRemovedInfo;
  }

  public tweetLocationRemovedInfo: ITweetLocationRemovedInfo;
}

export class TweetWitheldEventArgs /*extends EventArgs*/ {
  constructor(tweetWitheldInfo: ITweetWitheldInfo) {
   // super();

    this.tweetWitheldInfo = tweetWitheldInfo;
  }

  public tweetWitheldInfo: ITweetWitheldInfo;
}
